packages/ra-core/src/controller/edit/useEditController.ts

```diff
@@ -203,6 +212,9 @@ export const useEditController = <
                                   },
                         }
                     );
+                    if (shouldRedirectAfterUpdate) {
```

Comment:
You should add the same comment as for the useCreateController here, otherwise it's unclear why you need to do that.